### PR TITLE
33 logger

### LINKED ISSUE #33: Implement Logger
Implement a logger with two prority levels:
- Warnings that should be shown to the user
- Debug messages that should only be shown during development

### PR DESCRIPTION
**Pull request by mpsijm**
Will close #33 

I implemented a Logger class. It is used instead of the `console.log` calls. It adds some 5-character-long tags to our log, like `[WARN ]` or `[DEBUG]`. The `ConsoleLogDatabaseAdapter` also uses the logger and has the tag `[DATA ]`. 
The debug messages are not enabled by default. If you call `Logger.setDebug()` (either in code or in the console), the debug messages will be shown after that point.

Let your feedback flow over me! :smile:

### CODE REVIEW

File: src/octopeer-github/main/Logger.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Created by Maarten on 17-05-2016.
+ */
```
Comment (mdingjan) [inline src/octopeer-github/main/Logger.ts]: An extra descriptive line, just to keep the documentation sufficient? :)

**Comment by mdingjan** [comment]
I like the tags :D 

Maybe you would want to wait for my PR to be merged and merge it into your branch?
We could check the coverage changes and check whether it's needed to add more tests (just to be sure :P).

**Comment by mpsijm** [comment]
Let's merge the heck out of our branch then :D I reviewed it at least, but we need a few more :)

File: src/octopeer-github/main/Logger.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+const Logger = new (class Logger {
+
+    private isDebug = false;
+
+    public log(obj: any) {
```
Comment (thervh70) [inline src/octopeer-github/main/Logger.ts]: I would prefer if this would be refactored to `warn()`. Then you can refactor `_log()` to `log()`. That would give more clarity in your code.

**Comment by thervh70** [comment]
Looks good @mpsijm!

Merge approved.

File: src/octopeer-github/main/Logger.ts (hunk 1 above)
Comment (MathiasMeuleman) [inline src/octopeer-github/main/Logger.ts]: +1

File: src/octopeer-github/main/Logger.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Created by Maarten on 17-05-2016.
+ */
+
+const Logger = new (class Logger {
```
Comment (Exclaminator) [inline src/octopeer-github/main/Logger.ts]: Is this the way to create singletons?

**Comment by MathiasMeuleman** [comment]
Please fix the comments and merge devin this branch (it still contains conflicts). Then this seems ready to be merged

**Comment by MathiasMeuleman** [comment]
And all checks seem to fail all of a sudden :)

**Comment by Exclaminator** [comment]
Hmmm his is verry helpfull and a good adition, the test and all look nice, but please fix the failing build.

**Comment by mpsijm** [comment]
The failing tests seem to be caused because the main.ts files are also compiled into the test file. It's really odd that this didn't happen before, but I'm trying to fix it :)

**Comment by mpsijm** [comment]
I am missing one line of test coverage. The `database` method does the same as the `log` -> `warn` method, except with a different tag.

File: src/octopeer-github/main/Logger.ts (hunk 2 above)
Comment (mpsijm) [inline src/octopeer-github/main/Logger.ts]: Fixed!

File: src/octopeer-github/main/Logger.ts (hunk 1 above)
Comment (mpsijm) [inline src/octopeer-github/main/Logger.ts]: Added JSDoc to the entire Logger class, thanks for noticing :)

File: src/octopeer-github/main/Logger.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    private isDebug = false;
+
+    /**
+     * Logs a warning to the console.
+     * @param obj The object to be logged.
+     */
+    public warn(obj: any) {
+        this.log("WARN ", obj);
```
Comment (Exclaminator) [inline src/octopeer-github/main/Logger.ts]: Maybe extract these strings to contants, so they can be referd to from either context

Comment (mpsijm) [inline src/octopeer-github/main/Logger.ts]: Why is this necessary? I get your point, but because the strings are not (going to be) duplicated anywhere, I think this is not necessary.

**Comment by MathiasMeuleman** [comment]
Good fix, will merge
